### PR TITLE
Jormun: fix new default

### PR DESCRIPTION
**Pull request by xlqian**
http://jira.canaltp.fr/browse/NAVITIAII-2080 

Fix new default:
- Select better solutions
- Must-have journeys are now included in the candidates pool

### CODE REVIEW

**Comment by xlqian** [comment]
Jenkins, test this please!

**Comment by buildbotctp** [comment]
Build finished. No test results found.

**Comment by xlqian** [comment]
Jenkins, test this please!

File: source/jormungandr/jormungandr/scenarios/new_default.py [hunk 1]
```diff
@@ -330,20 +330,23 @@ def culling_journeys(resp, request):
...
 
-    nb_journeys_must_have = len(resp.journeys) - len(candidates_pool)
+    nb_journeys_must_have = len(idx_of_jrnys_must_keep)
     logger.debug("There are {0} journeys we must keep".format(nb_journeys_must_have))
-    nb_journeys_to_find = request["max_nb_journeys"] - nb_journeys_must_have
-    if nb_journeys_to_find <= 0:
-        # In this case, max_nb_journeys is smaller than nb_journeys_must_have, we have to make choices...
-        for jrny in candidates_pool:
-             journey_filter.mark_as_dead(jrny, 'Filtered by max_nb_journeys')
+    if request["max_nb_journeys"] == nb_journeys_must_have:
```
Comment (pbougue) [inline source/jormungandr/jormungandr/scenarios/new_default.py]: this is gonna do nothing when you should instead just keep the must_have journeys

**Comment by xlqian** [comment]
Jenkins, test this please!

File: source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py [hunk 2]
```diff
@@ -144,34 +152,41 @@ def create_candidate_pool_and_sections_set_test():
...
 def build_candidate_pool_and_sections_set_test():
     mocked_pb_response = build_mocked_response()
-    candidates_pool, sections_set =new_default._build_candidate_pool_and_sections_set(mocked_pb_response)
+    candidates_pool, sections_set, idx_jrny_must_keep = \
+        new_default._build_candidate_pool_and_sections_set(mocked_pb_response)
     selected_sections_matrix = new_default._build_selected_sections_matrix(sections_set, candidates_pool)
-    # selected_sections_matrix should have 13 lines(13 journeys) and 7 columns(7 sections)
-    assert selected_sections_matrix.shape == (13, 7)
+
+    # selected_sections_matrix should have 17 lines(17 journeys) and 11 columns(11 sections)
```
Comment (pbougue) [inline source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py]: comment not matching code (19 journeys)

File: source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py [hunk 3]
```diff
@@ -144,34 +152,41 @@ def create_candidate_pool_and_sections_set_test():
     Given response, the tested function should return a candidate pool and a section set
     """
     mocked_pb_response = build_mocked_response()
-    candidates_pool, sections_set =new_default._build_candidate_pool_and_sections_set(mocked_pb_response)
+    candidates_pool, sections_set, idx_jrny_must_keep = \
+        new_default._build_candidate_pool_and_sections_set(mocked_pb_response)
 
     # We got 17 journeys in all and 4 of them are tagged with 'best', 'comfort', 'non_pt_bike', 'non_pt_walk'
```
Comment (pbougue) [inline source/jormungandr/jormungandr/scenarios/tests/new_default_tests.py]: comment not matching code

**Comment by pbougue** [comment]
Also needs to integrate test for STIF bug (when requested journeys == must_have journeys)
cf. PR you were preparing.

**Comment by xlqian** [comment]
Jenkins, test this please!

**Pull request merged**